### PR TITLE
Lia random splitting

### PR DESCRIPTION
**Pull request by aehyvari**
The LIA solver, upon finding a solution to the linear system assigning non-integer values to integer variables, needs to communicate to the SAT solver how to fix a value to an integer.  Previously this was done by choosing the variable to be fixed by choosing the one furthest away from an integer.  It turns out that a completely random heuristic is better than the previous heuristic.  While we are trying to find a better heuristic, this PR suggests to switch to a completely random heuristic.

### CODE REVIEW

File: src/common/Random.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//
+// Created by Antti Hyvarinen on 07.05.22.
+//
```
Comment (blishko) [inline src/common/Random.h:3]: Use new license header :)

File: src/common/Random.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+#ifndef OPENSMT_RANDOM_H
+#define OPENSMT_RANDOM_H
+
+namespace opensmt {
+    // Returns a random float 0 <= x < 1. Seed must never be 0.
+    static inline double drand(double &seed) {
+        seed *= 1389796;
```
Comment (blishko) [inline src/common/Random.h:11]: Maybe add `assert(seed != 0)`?

File: src/tsolvers/lasolver/LASolver.h [hunk 3]
```diff
@@ -151,8 +152,8 @@ class LASolver: public TSolver
     LVRef getVarForTerm(PTRef ref) const  { return laVarMapper.getVarByPTId(logic.getPterm(ref).getId()); }
     void notifyVar(LVRef);                             // Notify the solver of the existence of the var. This is so that LIA can add it to integer vars list.
 
-    // Most-infeasible branching heuristic
-    LVRef splitOnMostInfeasible(vec<LVRef> const &) const;
+    // Random splitting heuristic
+    LVRef splitOnRandom(vec<LVRef> const &);
```
Comment (blishko) [inline src/tsolvers/lasolver/LASolver.h:156]:
```suggestion
    LVRef splitOnRandom(vec<LVRef> const &) const;
```
This can be `const`, right?

**Review by blishko — COMMENTED** [review]
LGTM!
Pulling the random functions to a separate header is a good thing!
There are just a few tiny things.

File: src/common/Random.h (hunk 1 above)
Comment (aehyvari) [reply src/common/Random.h:3]: Technically it's not my code, it's probably from MiniSat.  I'm not sure what to say there...  Should I put Minisat's license and Een & Sorenson as authors, addition to me?

File: src/common/Random.h (hunk 2 above)
Comment (aehyvari) [reply src/common/Random.h:11]: OK.

File: src/tsolvers/lasolver/LASolver.h (hunk 3 above)
Comment (aehyvari) [reply src/tsolvers/lasolver/LASolver.h:156]: It changes `LASolver::seed`.

**Comment by blishko** [comment]
@aehyvari, how about we use some standard C/C++ utilities for the random number generation? Is there any reason why we should stick to the functions from MiniSAT?

**Comment by blishko** [comment]
Or if that is too complicated, maybe we can stick to this for now. I don't know about the license in this case. If you want you can attribute it to the authors of MiniSAT. Although, then you could argue that we should do it for more files, such as `CoreSMTSolver`...

**Review by blishko — CHANGES_REQUESTED** [review]
The `#include Random.h` has been left in the `CoreSMTSolver.h`

File: src/smtsolvers/CoreSMTSolver.h [hunk 4]
```diff
@@ -62,6 +62,7 @@ OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWA
 #include "SolverTypes.h"
 
 #include "Timer.h"
+#include "Random.h"
```
[the remark this answers is not in the document]
Comment (aehyvari) [reply src/smtsolvers/CoreSMTSolver.h]: Ok.

**Review by blishko — APPROVED** [review]
LGTM!

**Pull request merged**